File: 0704-Binary-Search/0704-Binary-Search.java
class Solution {
    public static int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length-1 ;

        while (left <= right) {
            int middle = left +(right - left) / 2;
            System.out.println(middle);
            if (nums[middle] == target) {
                return middle;
            }
            if (target > nums[middle]) {
                left = middle + 1;
            } else {
                right = middle - 1;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] x = {9};
        System.out.println(search(x, 9));
    }
}

File: 0704-Binary-Search/0704-Binary-Search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left,right =0,len(nums)-1
        while left<=right:
            if nums[left + (right -left)//2]==target:
                return left + (right -left)//2
            elif target > nums[left + (right -left)//2]:
                left =( left + (right -left)//2)+1
            else :
                right = (left + (right -left)//2)-1
        return -1
